TigranKirakosov/functions.js

```diff
@@ -0,0 +1,66 @@
+// 1) splitAndMerge
+
+//version 1
+function splitAndMerge(str, sp) {
+  let acc = [];
+  let newStr = '';
+  for (let i = 0; i < str.length; i++) {
+    if (str[i] === ' ') {
+      acc[i] = str[i];
+    } else {
+      if (str[i + 1] === ' ' || i === str.length - 1) {
+        acc[i] = str[i];
+      } else {
+        acc[i] = str[i] + sp;
+      }
+    }
+    newStr += acc[i];
+  }
+  return newStr;
+}
+
+//version 2
+function splitAndMerge2(str, sp) {
+  let newStr = '';
+  str.split(' ').forEach((word, index, arr) => {

```

Comment:
you can avoid edge case for last element, if rewrite it to smth like
`str.split(' ')
    .map(function(word) { return word.split('').join(sp) })
    .reduce(function(resultSentence, modifiedWord) { return resultSentence + modifiedWord }, '')`

Reply:
Yeah, actually I have some improvements which weren't committed yet